src/aspire/storage/starfile.py

```diff
@@ -205,3 +205,43 @@ def __eq__(self, other):
                 if not self_list[i][1] == other_list[i][1]:
                     return False
         return True
+
+
+def getRelionStarFileVersion(filepath):
+    """
+    Determines whether a supplied STAR file is of Relion 3.0 or 3.1 and on
+        format.
+    :param filepath: Path to a STAR file.
+    :return: Either "3.0" or "3.1", or `None` if ASPIRE cannot interpret this file as a RELION
+        STAR file.
+    """
+    star = StarFile(filepath)
+    # 3.0 files have one block, >=3.1 files have two
+    if len(star) == 1:
+        data_block_columns = star.get_block_by_index(0).columns.to_list()
```

Comment:
Done, see comment above, thanks!

Reply:
Yeah, generally it doesn't have the name in the older files.. I could implement saving the `data_block` name for the 3.1 Starfiles?

Reply:
Yeah that would be good. I think the name is going to depend on the type of STAR file we're dealing with. If it's indexing micrographs, the block will probably be called `"images"`, if it's indexing single particle projections, it'll be `"particles"`, and so on.

Reply:
Going off the Relion 3.1 tutorial data,

in the Import job, it has movies in a block called data `movies`

in the CTFfind/other micrograph jobs, it has micrographs in a block called `micrographs`

And for the rest of the pipeline, generally `particles`.

Reply:
Also, sorry now I see you specified `Relion31StarFile` in your original comment :) working on it

Reply:
I had one additional question for you regarding this line. Given that the legacy STAR files usually have a block with blank title, do you think we'd want to access via `blocks[""]` or get the sole block by index, e.g. `star.get_block_by_index(0)` as I have above?

Reply:
For the legacy STAR files, I think this is less of an issue, since we know that there's only a single block. There it may make more sense to take the only block instead of worrying about names.